ch6/problem_2.py

```diff
@@ -1,2 +1,2 @@
 def to_generator(map_or_filter):
-    pass
+    return (x for x in map_or_filter)
```

Comment:
This doesn't quite answer the problem, which was to get the underlying function out of the map/filter using `.__reduce__()` and recreate the result.